src/domains/wallet-common/03_cosmosdb.tf

```diff
@@ -74,3 +74,120 @@ resource "azurerm_cosmosdb_mongo_database" "wallet" {
 }
 
 # Collections
+locals {
+  collections = [
+    {
+      name = "services"
+      indexes = [{
+        keys   = ["_id"]
+        unique = true
+        },
+        {
+          keys   = ["name"]
+          unique = true
+        }
+      ]
+      shard_key = null
+    },
+    {
+      name = "wallets"
+      indexes = [{
+        keys   = ["_id"]
+        unique = true
+        },
+        {
+          keys   = ["userId"]
+          unique = false
+        }
+      ]
+      shard_key = "_id"
+    },
+    {
+      name = "wallet-log-events"
+      indexes = [{
+        keys   = ["_id"]
+        unique = true
+        },
+        {
+          keys   = ["walletId", "timestamp", "eventType"]
+          unique = true
+        }
+      ]
+      shard_key = null
+    },
+  ]
+}
+
+module "cosmosdb_wallet_collections" {
+
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//cosmosdb_mongodb_collection?ref=v6.7.0"
+
+
+  for_each = {
+    for index, coll in local.collections :
+    coll.name => coll
+  }
+
+  name                = each.value.name
+  resource_group_name = azurerm_resource_group.cosmosdb_wallet_rg[0].name
+
+  cosmosdb_mongo_account_name  = module.cosmosdb_account_mongodb[0].name
+  cosmosdb_mongo_database_name = azurerm_cosmosdb_mongo_database.wallet[0].name
+
+  indexes     = each.value.indexes
+  shard_key   = each.value.shard_key
+  lock_enable = var.env_short == "d" ? false : true
```

Comment:
since we are refining the models/db, it is better if we enable the lock only in PROD

Reply:
You're right!